helm/prometheus-operator-app/values.yaml

```diff
@@ -193,6 +193,47 @@ prometheus-operator-app:
   kubeStateMetrics:
     enabled: true
   kube-state-metrics:
+    extraArgs:
+    - --custom-resource-state-config
+    # This will output the following metric:
+    # HELP kube_customresource_autoscaling_annotations Kubernetes annotations converted to Prometheus labels.
+    # TYPE kube_customresource_autoscaling_annotations gauge
+    # kube_customresource_autoscaling_annotations{customresource_group="autoscaling.k8s.io", customresource_kind="VerticalPodAutoscaler", customresource_version="v1", namespace="default",target_api_version="autoscaling.k8s.io/v1",target_kind="Deployment",target_name="hamster",verticalpodautoscaler="hamster-vpa"} 123
+    - |
+        spec:
+          resources:
+          - groupVersionKind:
+              group: autoscaling.k8s.io
+              kind: "VerticalPodAutoscaler"
+              version: "v1"
+            labelsFromPath:
+              verticalpodautoscaler: [metadata, name]
+              namespace: [metadata, namespace]
+              target_api_version: [apiVersion]
+              target_kind: [spec, targetRef, kind]
+              target_name: [spec, targetRef, name]
+            metricNamePrefix: kube_verticalpodautoscaler
+            metrics:
+              - name: "annotations"
+                help: "Kubernetes annotations converted to Prometheus labels."
```

Comment:
Annotations and labels can be high cardinality.
Is it something added here, or something that was in defaults and has to be explicitely set here to be kept?
Maybe there's a way to only keep a few specific labels/annotations we use?

Reply:
So I tried Yesterday to look at how we should configure this by keeping the default metrics, especially those that could bé useful to us around the status values but labels and annotations are thé only one that worked ... It's a bit challenging